### DISCUSSION TITLE
Themes: Ideas, Feedback, Accessibility, Discussion

### DISCUSSION THREAD

**Discussion by iamnbutler** [zed-industries/zed#6657] [Feature Requests]
Starting a discussion to share and get feedback on all things themes.

The docs page for themes can be found here: [docs.zed.dev/configuration/themes](https://docs.zed.dev/configuration/themes)

**Comment by iamnbutler** [comment]
One thing I want for Zed themes in the future is to be able to dynamically shift the colors of a theme. For example: Higher/lower contrast or saturation modes.

I'd love to eventually ship something like a colorblind filter (shift hue/saturation based on color blindness type – [Example](http://gameaccessibilityguidelines.com/wp-content/uploads/2016/06/destinycolorblind.jpg))

Something like this would be a ways off though.

**Reply by johnm** [reply]
Yes, so many themes are really washed out. 

I'm less interested in something automated to change those things and much more excited by the ability to quickly and easy fork an existing theme and tweak it to my preferences.

I don't even care if it's a gui-ish thing to edit theme, just that it's fast and easy to fork/tweak the colors. But I definitely won't say no to a decent gui. :-)

**Reply by johnm** [reply]
Big +1 to support the different types of color blindness. I think that would be very helpful during collaboration.

**Reply by etsea117** [reply]
As someone with Deuteranomaly, I would love a theme and/or a way to adjust themes to improve visibility. All the themes currently look like they've faded from being in the sun too long.

**Reply by ghost** [reply]
could an experimental feature be added for custom themes following the same theme formatting and syntax, etc. that is already used for the current themes that we want to use.. so maybe we can make and break and tinker with it to help... i know there is no time table yet for custom themes.. i know it would be extremely experimental yet maybe some of us could use it if u implemented something basic so we can do some of the legwork in our usual day to day as i have a feeling some of us will want to tinker anyways based on what i have read so far.. although.. we wouldn't have any support on it because until u got around to it we would be sort of data gathering and sharing as we learn to help.. idk.. random idea... maybe this is beneficial?..

**Comment by Daeraxa** [comment]
I'd love to see the barrier of entry to theming be lower than it often is so that people can easily create and share themes.
I really like the approach [Lapce has taken to theming](https://docs.lapce.dev/development/theme) with a simple GUI layout to select a colour against a list of different items which can then be exported and shared. I can easily see this being extended with a colour picker and some other nice things to make it really quite easy.

**Reply by iamnbutler** [reply]
Yeah we have lots of plans™️ around this, but there is a lot to consider, and we have to sequence priorities. These are some old designs that explore this idea: 

<img width="1391" alt="CleanShot - 2022-09-08 at 12 40 52@2x" src="https://user-images.githubusercontent.com/1714999/189178362-7a0f8967-ab41-4b82-813a-2c261ee5a5e0.png">

<img width="1328" alt="CleanShot - 2022-09-08 at 12 41 06@2x" src="https://user-images.githubusercontent.com/1714999/189178415-a9956052-2213-437f-b04c-32ab4273d124.png">

<img width="1174" alt="CleanShot - 2022-09-08 at 12 41 18@2x" src="https://user-images.githubusercontent.com/1714999/189178453-9fca8173-b6cd-487f-ad11-574d9b6bdec7.png">

**Reply by johnm** [reply]
> Yeah we have lots of plans™️ around this, but there is a lot to consider, and we have to sequence priorities. These are some old designs that explore this idea:

If/when you go down that rabbit hole, please do allow for the settings of the editor chrome to be separate/separable from the settings for the editing pieces.

**Reply by ghost** [reply]
can some of us be able to play with those settings while u are sequencing said priorities.. maybe we can help during?.. unless u are talking about having to hard code the GUI with it... to hold us off until then maybe a css based basic sytax for theme switching with a link within or something or having a page on the git here with the themes available.. at first i get there may be more legwork.. i  loved Atom myself and really wanna see this project take off! maybe this helps?

**Reply by pedrosouza458** [reply]
wow, i really like this, it should be great to have, good luck

**Reply by xtrasmal** [reply]
![super_meat_boy](https://github.com/user-attachments/assets/5dbe27df-4b6d-447c-b3d0-0656ccd7fd66)

**Comment by MatthiasGrandl** [comment]
I would like to request the option to set different themes for light/dark system preferences. My system theme preference changes throughout the day and I really appreciate it if my editor changes automatically with it.

**Reply by johnm** [reply]
Yes! I manually do this at least twice everyday.

**Reply by ghost** [reply]
or even a basic line of code to enable auto switching on a variable timer based on preferred personal input, that could be configured within the configuration file possibly... if that is helpful for the above idea to help streamline the thought and idea to practical application for real time usage..  had that thought.. maybe this helps the devs?..

**Reply by iamnbutler** [reply]
This isn't ready yet, but the system we are working on now will help enable this!

**Comment by Aeolun** [comment]
Aww, I was hoping to be able to create my own theme. Just exposing the underlying theme format is fine (I don't need a whole UI), but it's a bit sad to have themes without customization. Even Fleet has only light/dark, but at least you can change it.

To be honest, I'm not quite sure _why_ this is something that I think is indispensable.

**Reply by ghost** [reply]
it is mentioned that they are working on custom themes yet there is no time table..
<img width="411" alt="Screenshot 2023-01-02 at 10 09 39 AM" src="https://user-images.githubusercontent.com/120283763/210249544-1f166fa0-25bc-40fe-a81d-58a09f227f0f.png">

**Reply by LukasRennhofer** [reply]
I am developing a new Alternative for Zed with a few Friends . I will Release The Code Editor Soon and it ist Filly customizabke and easy to use and is 2 Times faster 🔥 You should check it out :https://github.com/LL-Program/Minsky

**Reply by Moshyfawn** [reply]
@LL-Program, please avoid spamming Zed community platforms with your promotional material.

**Comment by ghost** [comment]
<img width="679" alt="Screenshot 2023-01-02 at 10 27 24 AM" src="https://user-images.githubusercontent.com/120283763/210251525-a27d6905-c3b8-4a4e-be85-50b1d8589bf2.png">
the issue i have noticed and it may be something simple.. though it is infuriating.. when i try to add a line to config file it auto updates and reverts as if needed for formatting within a second... have tried removing my auto save i added.. is this a bug?..

**Reply by ghost** [reply]
i found it! okay the autosave feature with milliseconds forces auto indenting and makes it extremely painful to add lines to config file, may be a bug within the coded formula on back end?.. or just me... essentially by switching auto save to off the issue is resolved, yet that seems to suggest the autosave feature may be a tad buggy on the millisecond timer... TL::DR seems to cause the file to force indent for proper syntax which i get for the config file then does the said autosave.. ..  anyone else have issue?.. anyone know of solution?.. going to do auto save feature last i suppose on config file in meantime.. will update when i make it to my website i am coding to practice in once configured to see if issue persists beyond the config file itself.

**Reply by ghost** [reply]
another solution.. on_focus_change prevent said issue.. so there is that.. hope that helps devs on the timer setting for the autosave feature  via milliseconds.. so maybe add an else statement to the variable on back end so  when the auto save is done via timer it doesnt auto indent like the other variables then it wont be such a headache and feel pretty much useless unless given a ridiculous amount of time to save tho I forsee that even that would be an issue because u may be lines of code in and then it autosaves in the window and ur work is scrubbed at least on the line your working on... maybe this helps.?

**Reply by iamnbutler** [reply]
I'm not sure what this has to do with themes. You probably will have better luck posting this as a separate discussion or issue.

**Comment by resistor4u** [comment]
I'm aware it may be controversial, but I believe the best text editors are the ones that allow users to configure window transparency. I am not talking about cosmetic blurs, which IMHO are worthless wastes of CPU cycles. What I'm talking about is the main window to have a background view with an editable alpha level. For Macs, TextMate up to version 2.0-rc.13 enabled this, but it was removed in later versions. As I understand it, the feature was removed because the methods it implemented were deprecated or removed in more modern Mac SDKs. iTerm enables this feature still, and also does it very, very well. It's been on my TODO to look into exactly how iTerm does it. Mac's native Terminal.app also implements configurable window transparency.

It's an essential feature to me, and I cannot be alone. It's essential because laptop screen space is limited, and if you find yourself drafting documents, then it's an incredible time-saver to have different versions pulled up you can re-write without copy-pasting or breaking your stride to `cmd + tab` to the other window. I will have yesterday's draft pulled up and today's draft slightly overlapping it, so that I can see what I've written through the window of today's draft. Sublime Text apparently enables this for Windows users, but it's a no-go for Mac. I don't think Atom had this feature. BBEdit definitely does not have it. So, I've kept TextMate 2.0-rc.13 around, and have (superstitiously?) refrained from upgrading MacOS beyond 10.15.7 for fear that the transparency feature will break.

At a high design level, it makes sense to implement window transparency as theme that users can enable and tweak, if they like.

**Reply by johnm** [reply]
Yep, so many of these features (and, frankly, underlying philosophy) are why I'm still using Emacs after multiple decades.

**Reply by viniciuspalma** [reply]
Coming from iTerm and neovim, I also miss the opportunity to watch my videos or series at the same time that I code in one screen :D

**Reply by StuartMW-YL** [reply]
![Screenshot 2024-05-15 at 10 51 58 PM](https://github.com/zed-industries/zed/assets/164123258/637b9f24-0311-43c0-b8d6-352d5253cc8f)
Like this? (LHS: Zed, Centre: nvim, RHS: iTerm)

You can use the below to customise Zed's background:
```
experimental.theme_overrides": {
    "background.appearance": "blurred",
    "background": "#282A3650", 
```

**Comment by mtwebb** [comment]
Before I had access to the beta, I read the [docs](https://zed.dev/features), as such I started to associate the theme used there with zed.  I appreciate it's not the riot of color found in most themes.  I don't see it in the list of experimental themes (Andromeda is close).   I think the below theme is a good addition to the default set:
![multi-cursor-editing](https://user-images.githubusercontent.com/1376680/210694373-16700f27-aeb9-4a58-8443-f03005fa947b.jpg)

**Reply by xtrasmal** [reply]
yeah it's hard to find a balance between. I like less highlighting as well. 

Nuisance:
<img width="717" alt="image" src="https://github.com/user-attachments/assets/0d8c6e10-a0e1-493f-952f-7c54af1f85f5">

**Comment by sriranggd** [comment]
I also vote for a theme with higher contrast. The existing ones (especially the light themes) look a bit washed out and dull.

A light theme with slightly higher contrast would greatly help me in using Zed daily at work. I created a separate discussion about this here :  https://github.com/zed-industries/feedback/discussions/841 .

**Comment by SilentGlasses** [comment]
I think to start with, having the option to edit a CSS file of sorts where we can enter the primary set of colors ~would be a good start~.
I am using Warp Terminal at the moment and the way they do it is via a `Theme_Name.yaml` file where you can add HEX color codes like this:

```
terminal_colors:
  bright:
    green: "#20b96c"
    magenta: "#1c834f"
    red: "#1c7f4c"
    cyan: "#1db267"
    white: "#1ed174"
    black: "#183024"
    blue: "#1e9358"
    yellow: "#24ea8f"
  normal:
    red: "#1c7f4c"
    green: "#20b96c"
    cyan: "#1db267"
    white: "#1ed174"
    blue: "#1e9358"
    black: "#183024"
    yellow: "#24ea8f"
    magenta: "#1c834f"
background: "#13241b"
foreground: "#1ed174"
accent: "#1ed174"
details: "darker"
```

This would allow basic users to generate a new theme by only needing to apply their colors to one place.  I created a retro Green theme for Warp by just needing to select the right shades of green...
![Screen Shot 4](https://user-images.githubusercontent.com/22822565/214344087-0c0f84ad-63f6-4207-9f56-f58193fd4ae5.png)

**Comment by SilentGlasses** [comment]
If you will go with a proper UI option, I think so far from the screenshots shared, this is the most useful and appealing way to do it.
![Screen Shot 5](https://user-images.githubusercontent.com/22822565/214346963-38768885-9856-4334-b4cb-3e6032c224ca.png)

**Reply by ghost** [reply]
Suggestion: More padding or spacing between the 'enable night shift" and the color squares above it

**Comment by ghost** [comment]
**Question:** This is probably a dumb question, but is there a list of all themes in Zed? I checked the Themes and Configuration tabs [here](https://zed.dev/docs/configuration/configuring-zed)...

**Suggestion:** In the themes tab of https://zed.dev/docs/configuration/configuring-zed there should be a table of built-in Zed themes and maybe some featured community themes.

**Reply by JosephTLyons** [reply]
We don't have any community themes yet because we don't have a system in place to load in custom themes.  Themes are currently baked into Zed.  We plan to allow for custom themes in the future, we just haven't had the time to put the code in place.

There is a theme selector in Zed:

Command palette > "theme selector: toggle"

<img width="602" alt="SCR-20230218-ptu" src="https://user-images.githubusercontent.com/19867440/219904195-9dfedac3-a9a0-451e-999a-1e7140d3b456.png">

**Reply by ghost** [reply]
Gotcha, thanks!

**Reply by JosephTLyons** [reply]
Also, this the theme selector can be accessed via `cmd-k, cmd-t` (perform these as a sequence of commands in succession).  Not sure why we don't show this key command in the command palette next to the `theme selector: toggle` text...

**Reply by ghost** [reply]
I’d assumed that (VSCode has same settings) but wasn’t sure; I didn’t want to mess anything up and have to reinstall 😅

Like you said, I think it’d be helpful to add that to the Command Palette under the theme selector option.

**Reply by JosephTLyons** [reply]
Yeah, it should be there, that might be a bug.  I opened an issue for it.

**Reply by k11q** [reply]
> We don't have any community themes yet because we don't have a system in place to load in custom themes. Themes are currently baked into Zed. We plan to allow for custom themes in the future, we just haven't had the time to put the code in place.
> 
> There is a theme selector in Zed:
> 
> Command palette > "theme selector: toggle"
> 
> <img alt="SCR-20230218-ptu" width="602" src="https://user-images.githubusercontent.com/19867440/219904195-9dfedac3-a9a0-451e-999a-1e7140d3b456.png">

This is crucial, please put this in the docs ASAP. cc @iamnbutler

**Reply by ghost** [reply]
@khairulhaaziq +1

**Reply by iamnbutler** [reply]
Apologies, I wasn't aware finding themes was an issue people were having. There is an update to the docs deploying now that should explain clearly the ways of picking a theme!

**Comment by photex** [comment]
I think CLion / IntelliJ have the gold standard for theme setup. 

It’s stack based. You have General at the bottom, then "Language Defaults" (universal language concepts like keyword, string, number, and so on without any specifics), and finally Language specific things on top (letting you make customizations for syntax that isn't universally applicable). You can edit the scheme via the gui and you get a preview of the theme in the context you’re working with. 

Visual Studio might be an example of the absolute worst way to theme.

**Reply by ghost** [reply]
From what you’ve described, that sounds amazing!!!! I think Visual Studio Code has a decent theming system - live previews of themes in a web editor. Not to mention a third party web app for making themes! 😀

**Reply by photex** [reply]
I guess I should be more clear that I was referring to Visual Studio proper (or whatever we're saying to differentiate it from Code and ~MonoDevelop~ Visual Studio Mac)

**Reply by ghost** [reply]
Oh no I understood you were referring to the .NET IDE Visual Studio. From what I’ve seen of it, I agree with you; the theming sucks

**Reply by dmitriid** [reply]
The downside to IntelliJ is that it's very hard to figure out how to customise non-editor areas: tree views, gutters etc.

I'd split code and non-code in a theme, but have both of them as stacks as you described.

**Reply by darrenbarklie** [reply]
I'm not familiar with  CLion / IntelliJ setup (could you post a screenshot for reference @photex?), but I wonder if I'm imagining something similiar... For me, a palette of tones in a row of squares doesn't afford much context about the application of a theme.

I thought about presenting a "vertical slice" of directories, files, git status, editor windows, code blocks, syntax highlights, linting, with an editor window modal to allow real-time, contextual editing of most contexts within a single screen. Not unlike the VSCode live preview of themes, but with opportunity for realtime modification on top.

**Comment by georgemaine** [comment]
Would love to create or edit themes — just exposing the API would be enough while y'all figure out how to innovate in this area

**Comment by LoongLjt** [comment]


**Reply by Nicoka11** [reply]
Not the best place to ask but considering they gone for a completely native solution, with a custom renderer using rust it's probably not any time soon

**Comment by mhenrixon** [comment]
I would love to see something like this:

<img width="533" alt="CleanShot 2023-04-23 at 17 57 35@2x" src="https://user-images.githubusercontent.com/106795/233850548-4fec4781-9ac1-463a-b920-f407c4f7f020.png">

In Sublime Text, I have a plugin installed that handles color coding based on string length. This is very beneficial to how my brain works. The plugin can be fount at: https://github.com/mhenrixon/Sublime-Colorcoder

Curious if you'd be open to such a thing?

**Comment by k11q** [comment]
It isnt just the themes that are the problems but also syntax highlighting. I this this is the biggest thing stopping me from migrating to zed currently. I am so desperate to migrate from the bloated electron vs code

<img width="1680" alt="Screenshot 2023-05-01 at 7 37 34 AM" src="https://user-images.githubusercontent.com/101852870/235381330-eb4640d1-93d7-4a44-8597-12fe5b045be1.png">

**Reply by iamnbutler** [reply]
Just as an FYI (for the comment below this as well), Zed and VSCode will never match exactly from a highlighting perspective as they use different technologies under the hood. VSCode uses TextMate grammars while Zed uses Treesitter grammars.

It would be helpful to understand what specifically you are referring to here. Is it the rainbow brackets? Those specifically are not a feature of syntax highlighting but a standalone feature (which we do plan to build at some point)

**Comment by Nicoka11** [comment]
I hope the API will be available soon, I want to recreate the monokai theme suite, and like @khairulhaaziq I noticed that the syntax highlight isnt as complete, for example in TSX (and JSX) constant variables have the same colors as jsx tags
<img width="2048" alt="image" src="https://user-images.githubusercontent.com/75330147/236664679-c7a1a5a5-881a-4ced-8619-69ccf3e85b8f.png">

**Reply by katopz** [reply]
vote for `monokai`, only thing that block me to use `Zed` is a theme. My brain hurt when switching theme! 🫠

**Comment by diocletiann** [comment]
I would like to nominate https://github.com/savq/melange-nvim for addition to Zed. It's easily my favorite Vim theme and I've recreated it in JetBrains. Once I got used to scanning warm colors for control flow and cool colors for data it's hard to go back.

**Comment by shot-codes** [comment]
I can't live without Tonskys Alabaster theme. Everything else looks like an absolute mess after becoming accustomed to it. Also, the ability to color the text background instead of foreground is great for light color schemes (there's an example in the repo for Alabaster)

https://github.com/tonsky/sublime-scheme-alabaster

**Comment by apol76** [comment]
With the shell bottom pane using the same colors as the code on top of it, the two windows-parts are more or less indestinguishable for me. So for themes that come in pairs (light and dark variant), it would be great to have the upper part (code) use the dark theme of the pair, with the lower part (shell) use the light theme of the pair. Or vice versa.

I'm however not sure about the rest of the gui, I guess it should look like the shell pane, so that the code stands out from the rest of the window.

**Comment by mattly** [comment]
Seconding the request to add color to backgrounds as well as foregrounds -- it does help a lot with lighter themes. I still miss this feature from emacs or atom when using vscode.

**Comment by combdn** [comment]
Can we have an option to turn off syntax highlighting altogether? I’m unsure why, but my brain struggles to perceive the code if it’s colored like a Christmas tree. (In my own theme, I just use a dimmed color for the comments.)

It might be an unpopular thing, but I guess it is the simplest request to implement 😅

**Reply by shot-codes** [reply]
Have a look at Alabaster. I feel the same as you and once I discovered it I am unable to use anything else.

https://github.com/tonsky/sublime-scheme-alabaster

You can find ports for neovim and vscode if you are interested.

**Comment by overvale** [comment]
I really admire the work that's gone into the [Modus Themes](https://protesilaos.com/emacs/modus-themes). According to the docs:

> The Modus themes are designed for accessible readability. They conform with the highest standard for color contrast between combinations of background and foreground values. For small sized text, this corresponds to the WCAG AAA standard, which specifies a minimum rate of distance in relative luminance of 7:1.

Included are a few variations that target different kinds of color blindness. Seems like a worthy effort.

**Comment by koozz** [comment]
I like the [Nord theme](https://www.nordtheme.com), it's easy to [apply it to a lot of apps](https://www.nordtheme.com/ports) and have a unified color theme throughout your system.

**Comment by matthewgapp** [comment]
Is it just me or is it hard to see things in Zed? Something about the contrast or brightness makes it very hard to see, regardless of theme, which is one of the main reasons it's not my go-to editor yet.

**Reply by johnm** [reply]
Yeah, that drives me bonkers. At least now that we can start playing with the themes, I'll end up trying to override some values of some existing themes.

**Reply by overvale** [reply]
User Themes are now in [preview](https://zed.dev/blog/user-themes-now-in-preview). I've ported a few VSCode themes already.

**Comment by tecandrew** [comment]
finding it hard to customize the terminal text color to make it more readable like in VS Code's theme of Catppuccin

The highlighted line in Zed (right side) is hard to read with the light blue as the background

![image](https://github.com/zed-industries/zed/assets/21319730/ee4d1e1d-ef67-4275-a9a6-83094b8fabaa)

**Reply by tecandrew** [reply]
other difficult parameters to identify is the color of the correctly searched & highlighted text in the editor. can't seem to find the 'lightBlue' color in `self.context` in https://zed.dev/schema/themes/v0.1.0.json

![image](https://github.com/zed-industries/zed/assets/21319730/c15cc23b-06c4-4d64-82e9-699f50e91c1b)

in vscode's theming there are clear parameters for this

![image](https://github.com/zed-industries/zed/assets/21319730/e584d730-ef37-4659-80c0-60c15da4d076)

**Comment by noidwasavailable** [comment]
Where can I find the specs for the syntax object?

At `https://zed.dev/schema/themes/v0.1.0.json`, I can see:
```json
"syntax": {
    "description": "The styles for syntax nodes.",
    "default": {},
    "type": "object",
    "additionalProperties": {
        "$ref": "#/definitions/HighlightStyleContent"
    }
},
```

I have tried
- https://zed.dev/schema/themes/v0.1.0/definitions/HighlightStyleContent.json
- https://zed.dev/schema/themes/definitions/HighlightStyleContent/v0.1.0.json
- https://zed.dev/schema/themes/definitions/HighlightStyleContent.json
- https://zed.dev/schema/themes/v0.1.0/#/definitions/HighlightStyleContent.json
and others so far but no catch...!

**Comment by noidwasavailable** [comment]
I think a good addition to a theme generator workflow is to have a dummy file or a UI that shows you all the different parts of the selected theme. I really like on zed how I can open `zed: open default keymap` and see all the different options without having to worry about editing my default settings or something accidentally.

There are many aspects to a theme that are difficult to try out or test because they are not that common of an occurrence, such as `unreachable`, or key names which you are left guessing what they mean, such as `variable.special` or `string.special.symbol`.

Having a file, or even better, a piece of UI on a panel which shows you how different lines on your theme JSON affects how your editor looks would be a great UX to create your own themes.

**Comment by teziovsky** [comment]
Is there any guide on how to develop new themes in zed? 🤔 I mean step by step?

**Comment by jordy25519** [comment]
playing with an imported vscode theme via the zed tool (zed left, vscode right)

<img width="256" alt="Screenshot 2024-03-04 at 14 52 35" src="https://github.com/zed-industries/zed/assets/5133901/f4c87e06-60d5-4f1b-9f1e-3883b016f8b1">
<img width="254" alt="Screenshot 2024-03-04 at 14 52 30" src="https://github.com/zed-industries/zed/assets/5133901/0d1e7334-e469-488b-9d25-f2e813babb1a">

seems like vscode has more granular awareness of language syntax. see `pub,impl,fn,strct` all keywords same color while in vscode they differ

**Comment by james-gough-op** [comment]
Is there any ability to color folders and files yet?

Based on whether they've changed, are open etc.

**Comment by davccavalcante** [comment]
🟢 Brings the nostalgic essence of vintage green phosphor monitors to your Zed editor. This theme is crafted to offer a unique and immersive visual experience, capturing the original charm of early computing displays. Available in both dark and light versions, the Green Monochrome Monitor Theme transforms your coding environment with its distinct green-on-black and green-on-light-green aesthetics.

https://github.com/Takk8IS/green-monochrome-monitor-theme-for-zed

Green Monochrome Monitor Dark Theme
![screenshot-dark](https://github.com/zed-industries/zed/assets/152639968/fc8aecda-380a-4a27-8318-79c653304ad1)

Green Monochrome Monitor Light Theme
![screenshot-light](https://github.com/zed-industries/zed/assets/152639968/bf375857-4f65-4834-ae67-8f62ad479a42)

Experience the originality and unique aesthetics of the Green Monochrome Monitor Theme and take your coding journey back to the era of early computing displays.

**Reply by mattly** [reply]
this is not the place to submit themes, you can submit them [over here](https://github.com/zed-industries/extensions)

**Reply by davccavalcante** [reply]
Thank you very much for referring me to the appropriate place. =)

**Comment by andreasvirkus** [comment]
It'd be great if I could set/persist different themes per project, so I could more easily distinguish between the active window and get context quickly.

(Sorry if this isn't the correct discussion for it or if it's already possible today)

**Comment by Vfposselt** [comment]
Summercamp light theme, please!

This is pretty much anecdotal, but I think Summercamp is one the most original editor themes I've seen in the last couple of years. And as far as I'm aware, Summercamp is the only original Zed theme that was successfully ported to VSCode: https://github.com/csalmeida/summercamp-vscode-theme

With the introduction of default light theme in the stable release, it would be great to see a potential light version of it. As much as I think One Light and Dark are fine for default themes, they simply lack the originality that Zed provides.

**Comment by xtrasmal** [comment]
Options I'd like in theme.json:

Syntax nodes:
- font_features
- font_family
- more font_style options
- font_size

**Reply by the2nake** [reply]
Font family would be really nice for cursive italics and for viewing doc comments in a non-monospace font.

**Reply by zxaos** [reply]
For tracking, this was requested in https://github.com/zed-industries/zed/issues/4986 and in https://github.com/zed-industries/zed/issues/7218

Would be great to have.

**Comment by eike** [comment]
Three things from me:
- I would really like to be able to color in the *declarations* of variables (and functions, etc.), but not every single usage of them. This way, if there is a variable somewhere, it is much easier to scan for the place where it is defined.
- Playing around with a theme is very difficult right now because there is no way of seeing the theme-able scopes. Some mode or command that allows me to see which key to use in the theme JSON would be great. (Maybe I just haven't found it?)
- A way of adapting the theme to the language. Sometimes, one just doesn't agree that things in two different languages should be colored the same, but they happen to use the same theme key.

**Reply by mattly** [reply]
part of the problem for your second and third points is that the tokenization of things seems to vary greatly from grammar to grammar

for example, the bright purple italics in this screenshot is for `string.special.symbol`, which I didn't really see in any other language, but is used in Elixir for a few things and Erlang for _everything_. ![image](https://github.com/user-attachments/assets/9aaa273c-9938-47b3-8a31-89b58bf4a4bc)

I've seen a lot of other variations from language to language, Lua for example shows `{}` table literals as constructors. Getting a sense for what syntax selectors match which tokens is really hard when this varies so much from language to language.

**Reply by eike** [reply]
Yeah, that is why they would be nice. The second to figure this out, and the third to solve it.

**Comment by juliangarnier** [comment]
Something that I think would greatly improve accessibility and readability is the ability to define a foreground color for selected text, matches and highlighted text.

Currently, we are limited to using very subtle colors for highlighting text, which is counterintuitive for a highlight system IMO, and can results in readability issues.

Hard to read text in search results:
<img width="721" alt="Screenshot 2024-09-17 at 09 29 56" src="https://github.com/user-attachments/assets/938bc3df-826f-47ec-a90a-6cea71d82b7d">
<img width="626" alt="Screenshot 2024-09-17 at 09 57 50" src="https://github.com/user-attachments/assets/bd633349-bd81-48c2-9671-f2dce740d7b0">

Selecting a text can in some cases make it harder to read:
<img width="444" alt="Screenshot 2024-09-17 at 09 31 20" src="https://github.com/user-attachments/assets/284a7a8f-fcbc-4b51-b125-c82fe0b8e800">

Matching brackets are almost unnoticeable:
<img width="413" alt="Screenshot 2024-09-17 at 10 00 23" src="https://github.com/user-attachments/assets/0f9c43fd-e20f-45c3-b958-c87c6d03d99a">

## Suggested solution

Expose an optional foreground color property for each of the different match / highlight / selection styles:

```json
    "search.match_background": "#dfc184ff",
+   "search.match_foreground": "#282c34ff",
    "editor.document_highlight.read_background": "#dfc184ff",
+   "editor.document_highlight.read_foreground": "#282c34ff",
    "editor.document_highlight.write_background": "#dfc184ff",
+   "editor.document_highlight.write_foreground": "#282c34ff",
    "players": [
      {
        "cursor": "#5ac1feff",
        "background": "#5ac1feff",
        "selection": "#dfc184ff"
+       "selection_foreground": "#282c34ff"
      }
    ],
```

By setting high-contrast background and foreground colors to these properties, we make sure the text will alway be readable in all scenarios:

<img width="643" alt="Screenshot 2024-09-16 at 17 16 11" src="https://github.com/user-attachments/assets/68033b40-cb4c-4ced-8579-97a6921368dd">

(I already created an issue before stumbling on this discussion about this here: https://github.com/zed-industries/zed/issues/17890)

**Comment by AlvaroParker** [comment]
I don't know if this is the right place to discuss but given that fonts could be considered as part of the themes group I'll post it here. I think that the font on the diagnostic hover (triggered by `shift - k` when on vim mode) should use the `buffer_font_family` instead of the `ui_font_family` as it mostly displays code blocks or some kind of log, so it would be better that it uses the buffer font as this is usually monospace.

**Comment by erik-kroon** [comment]
is it possible to modify the color of nested object properties? 
<img width="297" alt="Screenshot 2024-10-18 at 10 26 29" src="https://github.com/user-attachments/assets/9af70e2e-3e9d-465c-b652-2e5f4dccbcd6">

**Comment by remixer-dec** [comment]
<img width="624" alt="screenshot" src="https://github.com/user-attachments/assets/5be34065-2abd-4162-94a1-c4a79e55db7d">
   
Just wanted to address poor (non-ai) completion quality, the order is random every time I start typing the word

**Comment by uloco** [comment]
As a theme auther, I'm looking for a way to show what highlight group / syntax group is applied to the word under cursor (Like :Inspect in Neovim treesitter) or "Inspect Editor Tokens and Scopes" in VSCode. How would I do this? There is not much documentation on how I could theme different languages for example, what scopes are themeable and provided by treesitter etc.

**Reply by tecandrew** [reply]
you can `cmd+shift+p` > `editor: copy highlight json` to copy the tokens & words extracted by tree-sitter for the current file you have open. paste the result into a new file and then comb through the results 

there's also `debug: open syntax tree view` menu

**Comment by apcamargo** [comment]
The red and green colors used in Git's diff visualizer are difficult to distinguish for people with colourblindness

![image](https://github.com/user-attachments/assets/b1a82d4e-dcd7-4f0b-a7c0-c65abdc1cc65)

**Comment by andrewjohnharvey** [comment]
I would love for you to add support for being able to set a font on a syntax level in themes. This is supported in the main themes but would be great for us to configure so that we can use something like [Monaspace](https://monaspace.githubnext.com/)

**Comment by aybarsnazlica** [comment]
The color of the comments used with the default One Dark theme has low contrast, which makes it hard to see the commented out text. It would be much easier to see the text if it's set to a brighter color.

The default look:

<img width="612" height="217" alt="Screenshot 2025-08-19 at 23 59 52" src="https://github.com/user-attachments/assets/1b9316f3-a2bf-4ccd-b8c9-69cc7b636ac8" />

After setting the color of "comment.doc"  to the color of "comment" in settings:

```json
"experimental.theme_overrides": {
    "syntax": {
      "comment": {
        "color": "#878e98" // changed
      }
    }
  }
```
<img width="612" height="217" alt="Screenshot 2025-08-20 at 0 00 10" src="https://github.com/user-attachments/assets/eee98039-8da4-481c-a61c-299c86b49348" />

**Comment by franciskafyi** [comment]
Hi everyone, I'm going to go ahead and close this as stale. We're not monitoring existing catch all discussions like this one well, so please open a new discussion with your specific request and we'll take a closer look 🙏
